pandas/io/formats/format.py

```diff
@@ -1553,7 +1554,7 @@ def _is_dates_only(
 
 def _format_datetime64(
     x: Union[NaTType, Timestamp],
-    tz: Optional[Union[tzfile, tzutc]] = None,
+    tz: Optional[Union["tzfile", "tzutc"]] = None,
```

Comment:
its not obvious to me why these only care about dateutil tzinfos and not e.g. stdlib or pytz versions.  @simonjayhawkins any idea?

Reply:
These were the types seen by MonkeyType. I think they are resolving to Any.

i've since abandoned using MonkeyType for a couple of reasons..

MonkeyType uses nominal types and in many cases these resolve to Any due to unfollowed imports.

MonkeyType only adds nominal types whereas we'd probably prefer structural types.

feel free to change or remove. The order of typing priority from high to low should probably match the order used in isort. so I consider these to be low priority type hints.

Reply:
OK.  I would be shocked if these weren't supposed to be `tzinfo`, will look more closely and update.

Reply:
yes, but if tzinfo is an unfollowed import, it'll just resolve to Any.

Reply:
what determines whether something is an unfollowed import?  tzinfo is stdlib, seems like mypy should know what it is

Reply:
> These were the types seen by MonkeyType. I think they are resolving to Any.

Revealed type for `tz` is `Union[Any, dateutil.tz.tz.tzutc, None]`

so it is only the `dateutil.zoneinfo.tzfile` that is unfollowed.

> its not obvious to me why these only care about dateutil tzinfos and not e.g. stdlib or pytz versions.

`pytz.tzinfo.DstTzInfo` is also unknown to mypy.

> tzinfo is stdlib, seems like mypy should know what it is

since all three (`dateutil.zoneinfo.tzfile`, `dateutil.tz.tz.tzutc` and `pytz.tzinfo.DstTzInfo`) inherit from `datetime.tzinfo`, and `datetime.tzinfo` is known to mypy through the stdlib, then it probably does make sense to use `datetime.tzinfo` here.

However, within the function, `tz` is only used in `Timestamp(x).tz_convert(tz)` and `Timestamp(x).tz_localize(tz)` and Timestamp resolves to any due to an unfollowed import. (needs stub #28019) so no actual type checking is being performed.

so probably best just to delete the type hint for now.